lib/utils/modules-to-certificates.js

```diff
@@ -0,0 +1,39 @@
+const { omit } = require('lodash');
+const crypto = require('crypto');
+
+/* eslint-disable camelcase */
+const createCertificateHash = ({certificate_number, accrediting_body, pass_date}) => {
+  const hash = crypto.createHash('sha256');
+  hash.update(certificate_number + accrediting_body + pass_date);
+  return hash.digest('hex');
+};
+/* eslint-enable camelcase */
+
+module.exports = function(modules) {
+  let certificates = [];
+
+  modules.forEach(moduleData => {
+    let moduleId = moduleData.id;
+    let moduleName = moduleData.module;
+    let certificateHash = createCertificateHash(moduleData);
+
+    let certificateIndex = certificates.findIndex(certificate => {
+      return certificate.hash === certificateHash;
+    });
+
+    if (certificateIndex !== -1) {
+      certificates[certificateIndex].modules.push({ id: moduleId, name: moduleName });
+      return;
+    }
+
+    let certificate = Object.assign({}, omit(moduleData, [
+      'id', 'module', 'migrated_id', 'species', 'not_applicable', 'other_accrediting_body'
+    ]));
+
+    certificate.hash = certificateHash;
+    certificate.modules = [{ id: moduleId, name: moduleName }];
+    certificates.push(certificate);
+  });
```

Comment:
Couple of points here:

* you can still use `const` for arrays and objects that have their properties modified, or get `push`ed to. `let` vs `const` only _really_ applies to reassignment, not modification.
* a `map` (to set the hash) and a `groupBy` (to group according to hash) is probably a clearer and cleaner way of writing this code.

Reply:
This _should_ roughly work:

```js
modules = modules.map(module => {
  return {
    ...module,
    hash: createCertificateHash(module)
  };
});
return groupBy(modules, 'hash');
```